**Linux with security manager** [Build 3209](http://brontes.lab.eng.brq.redhat.com/viewLog.html?buildId=73720&buildTypeId=WF_PullRequest_LinuxSm) outcome was **FAILURE** using a merge of 191f500ddbfaf8c0a18b319c540c188ca23880e4
Summary: Tests failed: 2 (2 new), passed: 938, ignored: 297 Build time: 00:10:32
### Failed tests
```
org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperationWithLegacyEntries: java.lang.AssertionError: WFLYMSG0081: Migration failed, see results for more details.
	at org.jboss.as.model.test.ModelTestUtils.checkOutcome(ModelTestUtils.java:121)
	at org.jboss.as.subsystem.test.AbstractSubsystemTest.checkOutcome(AbstractSubsystemTest.java:206)
	at org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperation(MigrateTestCase.java:155)
	at org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperationWithLegacyEntries(MigrateTestCase.java:134)
------- Stdout: -------
response = {
    "outcome" => "failed",
    "result" => {
        "migration-warnings" => [
            "WFLYMSG0082: Classes providing the remoting-incoming-interceptors are discarded during the migration. To use them in the new messaging-activemq subsystem, you will have to extend the Artemis-based Interceptor.",
            "WFLYMSG0082: Classes providing the remoting-outgoing-interceptors are discarded during the migration. To use them in the new messaging-activemq subsystem, you will have to extend the Artemis-based Interceptor.",
            "WFLYMSG0083: Could not migrate the HA configuration of [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\")
]. Its shared-store and backup attributes holds expressions and it is not possible to determine unambiguously how to create the corresponding ha-policy for the messaging-activemq's server.",
            "WFLYMSG0084: Could not migrate attribute local-bind-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-port from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupD\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-port from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupD\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0085: Could not create a legacy-connection-factory based on connection-factory [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"connection-factory\" => \"InVmConnectionFactory\")
]. It used a HornetQ in-vm connector that is not compatible with Artemis in-vm connector ",
            "WFLYMSG0080: Could not migrate attribute local-bind-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"broadcast-group\" => \"groupA\")
]. Use instead the socket-attribute to configure this broadcast-group.",


org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperationWithoutLegacyEntries: java.lang.AssertionError: WFLYMSG0081: Migration failed, see results for more details.
	at org.jboss.as.model.test.ModelTestUtils.checkOutcome(ModelTestUtils.java:121)
	at org.jboss.as.subsystem.test.AbstractSubsystemTest.checkOutcome(AbstractSubsystemTest.java:206)
	at org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperation(MigrateTestCase.java:155)
	at org.jboss.as.messaging.test.MigrateTestCase.testMigrateOperationWithoutLegacyEntries(MigrateTestCase.java:129)
------- Stdout: -------
response = {
    "outcome" => "failed",
    "result" => {
        "migration-warnings" => [
            "WFLYMSG0082: Classes providing the remoting-incoming-interceptors are discarded during the migration. To use them in the new messaging-activemq subsystem, you will have to extend the Artemis-based Interceptor.",
            "WFLYMSG0082: Classes providing the remoting-outgoing-interceptors are discarded during the migration. To use them in the new messaging-activemq subsystem, you will have to extend the Artemis-based Interceptor.",
            "WFLYMSG0083: Could not migrate the HA configuration of [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\")
]. Its shared-store and backup attributes holds expressions and it is not possible to determine unambiguously how to create the corresponding ha-policy for the messaging-activemq's server.",
            "WFLYMSG0084: Could not migrate attribute local-bind-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-port from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupC\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupD\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0084: Could not migrate attribute group-port from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"discovery-group\" => \"groupD\")
]. Use instead the socket-attribute to configure this discovery-group.",
            "WFLYMSG0080: Could not migrate attribute local-bind-address from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"broadcast-group\" => \"groupA\")
]. Use instead the socket-attribute to configure this broadcast-group.",
            "WFLYMSG0080: Could not migrate attribute local-bind-port from resource [
    (\"subsystem\" => \"messaging-activemq\"),
    (\"server\" => \"default\"),
    (\"broadcast-group\" => \"groupA\")
]. Use instead the socket-attribute to configure this broadcast-group.",


```

--- a/legacy/messaging/src/test/resources/org/jboss/as/messaging/test/subsystem_migration.xml
+++ b/legacy/messaging/src/test/resources/org/jboss/as/messaging/test/subsystem_migration.xml
@@ -121,8 +121,8 @@
                 <socket-binding>group-s-binding</socket-binding>
             </broadcast-group>
             <broadcast-group name="groupT">
-                <jgroups-stack>${jgroups.stack:udp}</jgroups-stack>
-                <jgroups-channel>${jgroups.channel:hq-cluster}</jgroups-channel>
+                <jgroups-stack>udp</jgroups-stack>
+                <jgroups-channel>hq-cluster</jgroups-channel>
             </broadcast-group>
         </broadcast-groups>
         <discovery-groups>
@@ -141,8 +141,8 @@
                 <socket-binding>group-t-binding</socket-binding>
             </discovery-group>
             <discovery-group name="groupU">
-                <jgroups-stack>${jgroups.stack:udp}</jgroups-stack>
-                <jgroups-channel>${jgroups.channel:hq-cluster}</jgroups-channel>
+                <jgroups-stack>udp</jgroups-stack>
+                <jgroups-channel>hq-cluster</jgroups-channel>
             </discovery-group>
         </discovery-groups>
         <diverts>

--- a/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java
+++ b/messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/CommonAttributes.java
@@ -175,7 +175,8 @@ public interface CommonAttributes {
 
     SimpleAttributeDefinition JGROUPS_STACK = create("jgroups-stack", ModelType.STRING)
             .setAllowNull(true)
-            .setAllowExpression(true)
+            // do not allow expression as this may reference another resource
+            .setAllowExpression(false)
             .setAlternatives("socket-binding",
                     "group-address", "group-port",
                     "local-bind-address", "local-bind-port")
@@ -184,7 +185,8 @@ public interface CommonAttributes {
 
     SimpleAttributeDefinition JGROUPS_CHANNEL = create("jgroups-channel", ModelType.STRING)
             .setAllowNull(true)
-            .setAllowExpression(true)
+            // do not allow expression as this may reference another resource
+            .setAllowExpression(false)
             .setAlternatives("socket-binding",
                     "group-address", "group-port",
                     "local-bind-address", "local-bind-port")

--- a/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_1_0.xml
+++ b/messaging-activemq/src/test/resources/org/wildfly/extension/messaging/activemq/subsystem_1_0.xml
@@ -162,8 +162,8 @@
         </acceptor>
 
         <broadcast-group name="groupT"
-                         jgroups-stack="${jgroups.stack:udp}"
-                         jgroups-channel="${jgroups.channel:activemq-cluster}"
+                         jgroups-stack="udp"
+                         jgroups-channel="activemq-cluster"
                          broadcast-period="${broadcast.group.period:1234}"
                          connectors="http netty"/>
         <broadcast-group name="groupS"
@@ -172,8 +172,8 @@
         <discovery-group name="groupT"
                          socket-binding="group-t-binding"/>
         <discovery-group name="groupU"
-                         jgroups-stack="${jgroups.stack:udp}"
-                         jgroups-channel="${jgroups.channel:activemq-cluster}"
+                         jgroups-stack="udp"
+                         jgroups-channel="activemq-cluster"
                          refresh-timeout="${discovery.group.refresh.timeout:2345}"
                          initial-wait-timeout="${discovery.group.initial.wait.timeout:2345}"/>
 

